locals.tf

```diff
@@ -10,6 +10,7 @@ locals {
   cluster_name              = "${var.prefix}-eks"
   default_tags              = { project_name = "viya" }
   tags                      = var.tags == null ? local.default_tags : length(var.tags) == 0 ? local.default_tags : var.tags
+  aws_shared_credentials    = var.aws_shared_credentials_file == "" ? var.aws_shared_credentials_files : [var.aws_shared_credentials_file]
```

Comment:
Might want to also check for a null values for both variables. If they either/both are set to null then things break.

Reply:
thanks, the older variable has a default of "", while the new one's default is null. Will need to guard against the user setting the old varaible to null as well.

Reply:
@thpang, if I try setting the deprecated `aws_shared_credentials_file` variable to null, Terraform identifies that as an error for calling length on a null value and halts. Are you suggesting to add a check with an error message if we detect a null before that can happen? The other new variable defaults to null already. So if we check for null, what action would you recommend us taking based on that check.

Reply:
You would need to have the following check:
```yaml
  # aws_shared_credentials_file - DEPRECATED
  aws_shared_credentials_file = var.aws_shared_credentials_file != null ? length(var.aws_shared_credenetials_file) > 0 ? true : false
  # Assign correct credential file value - If the old value is false, then new value must be used. Logic elsewhere would need to verify that check
  aws_shared_credentials = local.aws_shared_credentials_file ? [var.aws_shared_credentials_file] : var.aws_shared_credentials_files
```

Reply:
@thpang, I made the updates we discussed last week and also added a precondition check requiring var.aws_shared_credentials_file != null, the existing invalid function argument message will now be preceded by a fatal error message indicating that var.aws_shared_credentials_file must not be null as shown below. Also presents another opportunity to say that var has been deprecated and that aws_shared_credentials_files is now preferred.

> ╷
│ Error: Module output value precondition failed
│
│   on outputs.tf line 154, in output "aws_shared_credentials_file":
│  154:     condition = var.aws_shared_credentials_file != null
│     ├────────────────
│     │ var.aws_shared_credentials_file is null
│
│ aws_shared_credentials_file must not be null. aws_shared_credentials_file has been deprecated and will be removed in a
│ future release, use aws_shared_credentials_files instead.
╵
╷
│ Error: Invalid function argument
│
│   on outputs.tf line 162, in output "aws_shared_credentials":
│  162:     condition = length(var.aws_shared_credentials_file) == 0 || var.aws_shared_credentials_files == null
│     ├────────────────
│     │ while calling length(value)
│     │ var.aws_shared_credentials_file is null
│
│ Invalid value for "value" parameter: argument must not be null.
╵